src/objects.c

```diff
@@ -557,32 +558,26 @@ CK_RV p11prov_obj_find(P11PROV_CTX *provctx, P11PROV_SESSION *session,
     if (ret != CKR_OK) {
         return ret;
     }
-    while (total < MAX_OBJS_IN_STORE) {
-        CK_OBJECT_HANDLE object[OBJS_IN_SEARCH];
-        ret = p11prov_FindObjects(provctx, sess, object, OBJS_IN_SEARCH,
-                                  &objcount);
+    do {
+        objcount = 0;
+        objects = OPENSSL_realloc(objects, (total + OBJS_PER_SEARCH)
+                                               * sizeof(CK_OBJECT_HANDLE));
```

Comment:
does the `OPENSSL_realloc` handle freeying of the old location in case of failure? My reading of the source of openssl is that it just defers to standard `realloc` and it has the following note in the manual pages:

> If realloc() fails, the original block is left untouched; it is not freed or moved.

so on error, the original location needs to be freed too.

Reply:
true, I generally do that but forgot in this case